ecosystem/typescript/sdk/src/aptos_client.ts

```diff
@@ -453,11 +456,33 @@ export class AptosClient {
       await sleep(1000);
       count += 1;
     }
-    throw new WaitForTransactionError(`Waiting for transaction ${txnHash} timed out!`, lastTxn);
+    if (isPending) {
+      throw new WaitForTransactionError(`Waiting for transaction ${txnHash} timed out`, lastTxn);
```

Comment:
nit: can we also mention how long the waiting is in the message? something like `Waiting for transaction ${txnHash} timed out after 10s`